### PR TITLE
wpc-421 adding graphql description for public queries and mutations

### PR DESCRIPTION
**Pull request by HanaAyman**
This pull request is to add description to the graphQL public mutations and queries.

### CODE REVIEW

File: packages/api/src/graphql/mutation.public.ts [hunk 1]
```diff
@@ -261,6 +273,8 @@ export const GraphQLPublicMutation = new GraphQLObjectType<undefined, Context>({
       args: {
         email: {type: GraphQLNonNull(GraphQLString)}
       },
+      description:
+        "This mutation allows to reset the password by sending the user's email and a login link will be sent to that email.",
```
Comment (penk85) [inline packages/api/src/graphql/mutation.public.ts:277]: Maybe better worded:  "This mutation allows to reset the password by accepting the user's email and sending a login to that email.",

File: packages/api/src/graphql/mutation.public.ts [hunk 2]
```diff
@@ -316,6 +332,8 @@ export const GraphQLPublicMutation = new GraphQLObjectType<undefined, Context>({
         password: {type: GraphQLNonNull(GraphQLString)},
         passwordRepeated: {type: GraphQLNonNull(GraphQLString)}
       },
+      description:
+        "This mutation allows to update the user's password by entering the new password and the repeated new password and gives an error if the passwords don't match or if the user is not authenticated",
```
Comment (penk85) [inline packages/api/src/graphql/mutation.public.ts:336]: Maybe better in a split sentence:
"This mutation allows to update the user's password by entering the new password. The repeated new password gives an error if the passwords don't match or if the user is not authenticated."

File: packages/api/src/graphql/query.public.ts [hunk 3]
```diff
@@ -63,6 +64,7 @@ export const GraphQLPublicQuery = new GraphQLObjectType<undefined, Context>({
     peer: {
       type: GraphQLPeer,
       args: {id: {type: GraphQLID}, slug: {type: GraphQLSlug}},
+      description: 'This query takes either the ID or the slug and returns the peer profile.',
```
Comment (penk85) [inline packages/api/src/graphql/query.public.ts:67]: Just a note that you sometimes write ID lowercase, and probably best to be consistent.

File: packages/api/src/graphql/query.public.ts [hunk 4]
```diff
@@ -134,6 +139,7 @@ export const GraphQLPublicQuery = new GraphQLObjectType<undefined, Context>({
         slug: {type: GraphQLSlug},
         token: {type: GraphQLString}
       },
+      description: 'This query takes either the ID, slug ot token and returns the article.',
```
Comment (penk85) [inline packages/api/src/graphql/query.public.ts:142]: ot -> typo

File: packages/api/src/graphql/mutation.public.ts [hunk 5]
```diff
@@ -47,6 +47,8 @@ export const GraphQLPublicMutation = new GraphQLObjectType<undefined, Context>({
         email: {type: GraphQLNonNull(GraphQLString)},
         password: {type: GraphQLNonNull(GraphQLString)}
       },
+      description:
+        "This mutation allows to create a user session by taking the user's credentials email and password as an input and returns a session with token",
```
Comment (penk85) [inline packages/api/src/graphql/mutation.public.ts:51]: For consistency sake probably best to end sentences with a .

File: packages/api/src/graphql/mutation.public.ts [hunk 6]
```diff
@@ -101,6 +107,7 @@ export const GraphQLPublicMutation = new GraphQLObjectType<undefined, Context>({
     revokeActiveSession: {
       type: GraphQLNonNull(GraphQLBoolean),
       args: {},
+      description: 'This mutation checks if the session is authenticated and returns a boolean.',
```
Comment (roosnic1) [inline packages/api/src/graphql/mutation.public.ts:110]: this mutation **revokes** the active session. The sessions gets deleted.

File: packages/api/src/graphql/mutation.public.ts [hunk 7]
```diff
@@ -261,6 +273,8 @@ export const GraphQLPublicMutation = new GraphQLObjectType<undefined, Context>({
       args: {
         email: {type: GraphQLNonNull(GraphQLString)}
       },
+      description:
+        "This mutation allows to reset the password by accepting the user's email and sending a login to that email.",
```
Comment (roosnic1) [inline packages/api/src/graphql/mutation.public.ts:277]: ...sending a login link to...

**Review by roosnic1 — CHANGES_REQUESTED** [review]
looks good. I commented some tiny change requests

**Review by penk85 — APPROVED** [review]
:+1:

**Review by bishoymelek — APPROVED** [review]
:tada:

**Review by roosnic1 — APPROVED** [review]
🎉

**Pull request merged**